serde-support/src/main/java/io/micronaut/serde/support/serializers/CustomizedObjectSerializer.java

```diff
@@ -103,6 +104,22 @@ public void serializeInto(Encoder encoder, EncoderContext context, Argument<? ex
                     default:
                         // fall through
                 }
+                if (property.include != SerdeConfig.SerInclude.ALWAYS && property.include != SerdeConfig.SerInclude.NON_NULL) {
+                    switch (property.includeContent) {
```

Comment:
I don't think jackson does it this way

Reply:
I think it's only maps: https://www.javadoc.io/static/com.fasterxml.jackson.core/jackson-annotations/2.18.2/com/fasterxml/jackson/annotation/JsonInclude.html#content--

Reply:
Optional too

Reply:
Removed the common methods and logic and moved it to the optional

Reply:
It's a bit strange that you cannot exclude empty values from List